cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define all legal items and values for cylc suite definition files."""
+
+import os
+import re
+
+from metomi.isodatetime.data import Calendar
+
+from cylc.flow import LOG
+from cylc.flow import __version__ as CYLC_VERSION
+from cylc.flow.network.authorisation import Priv
+from cylc.flow.parsec.config import ParsecConfig
+from cylc.flow.parsec.upgrade import upgrader
+from cylc.flow.parsec.validate import (
+    DurationFloat, CylcConfigValidator as VDR, cylc_config_validate)
+from cylc.flow.hostuserutil import get_user_home, is_remote_user
+
+# Nested dict of spec items.
+# Spec value is [value_type, default, allowed_2, allowed_3, ...]
+# where:
+# - value_type: value type (compulsory).
+# - default: the default value (optional).
+# - allowed_2, ...: the only other allowed values of this setting (optional).
+SPEC = {
+    # suite
+    'process pool size': [VDR.V_INTEGER, 4],
+    'process pool timeout': [VDR.V_INTERVAL, DurationFloat(600)],
+    # client
+    'disable interactive command prompts': [VDR.V_BOOLEAN, True],
+    # suite
+    'run directory rolling archive length': [VDR.V_INTEGER, -1],
+    # suite-task communication
+    'task messaging': {
+        'retry interval': [VDR.V_INTERVAL, DurationFloat(5)],
+        'maximum number of tries': [VDR.V_INTEGER, 7],
+        'connection timeout': [VDR.V_INTERVAL, DurationFloat(30)],
+    },
+    'meta': {
+        'description': [VDR.V_STRING, ''],
+        'group': [VDR.V_STRING, ''],
+        'title': [VDR.V_STRING, ''],
+        'URL': [VDR.V_STRING, ''],
+        '__MANY__': [VDR.V_STRING, ''],
+    },
+    'general': {
+        'UTC mode': [VDR.V_BOOLEAN, False],
+        'cycle point format': [VDR.V_CYCLE_POINT_FORMAT],
+        'cycle point num expanded year digits': [VDR.V_INTEGER, 0],
+        'cycle point time zone': [VDR.V_CYCLE_POINT_TIME_ZONE],
+        'required run mode': [
+            VDR.V_STRING, '', 'live', 'dummy', 'dummy-local',
+            'simulation'
+        ],
+        'force run mode': [
+            VDR.V_STRING, '', 'live', 'dummy', 'dummy-local',
+            'simulation'
+        ],
```

Comment:
Not sure - I'd like another opinion - @oliver-sanders @hjoliver

Reply:
It might be nicer to face some of these topics at a later date when we can spare a little more time to rationalising the interface rather than chucking everything into this issue.

> Do we need 2 settings (and a command line option) for the run mode?

I'd do a quick grep to see how many suites are using the `force run mode` setting, my suspission is you'll return empty handed.

Reply:
I'm going to leave this for now. I think @Oliver-Sanders is right about this.

Reply:
Three tests are using "force run mode"; happy to get rid of it (in this PR or otherwise).